src/js/ripple/seed.js

```diff
@@ -57,104 +62,105 @@ Seed.prototype.parse_passphrase = function(j) {
-*
-*        {Uint160} (from_json able), specifies the address matching the KeyPair
-*                                    that is desired.
+Seed.prototype.get_key = function(opts) {
+  if (opts !== undefined && typeof opts !== 'object') {
+    throw new Error('get_key options not supported: ' + opts);
+  }
+  return this._get_key(opts);
+};
+
+/**
+* @return {KeyPair} - the root key-pair, as used by validators.
+*/
+Seed.prototype.get_root_key = function() {
+  return this._get_key({root: true});
+};
+
+/**
+* @param {Object} [options] -
 *
-* @param maxLoops (optional)
-*        {Number}                    specifies the amount of attempts taken
-*                                    to generate a matching KeyPair
+* @param {Number} [options.accountIndex=0] - the account number to generate
 *
+* @param {Boolean} [options.root=false] - generate root key-pair,
+*                                         as used by validators.
+* @return {KeyPair} -
 */
-Seed.prototype.get_key = function(account, maxLoops) {
-  var account_number = 0, address;
-  var max_loops = maxLoops || 1;
+Seed.prototype._get_key = function(options) {
+  var opts = options || {};
+
+  var root = opts.root;
 
   if (!this.is_valid()) {
     throw new Error('Cannot generate keys from invalid seed!');
   }
-  if (account) {
-    if (typeof account === 'number') {
-      account_number = account;
-      max_loops = account_number + 1;
-    } else {
-      address = UInt160.from_json(account);
-    }
-  }
 
-  var private_gen, public_gen;
+  var privateGen, publicGen;
   var curve = this._curve;
   var i = 0;
 
   do {
-    private_gen = sjcl.bn.fromBits(
-            firstHalfOfSHA512(append_int(this.to_bytes(), i)));
-    i++;
-  } while (!curve.r.greaterEquals(private_gen));
+    // We hash the seed to extend from 128 bits to 256, looping until we are
+    // sure the 256 bits represents a number that is situated on the curve.
+    privateGen = new Sha512().add(this.to_bytes()).addU32(i).finish256BN();
 
-  public_gen = curve.G.mult(private_gen);
+    // This private generator, represents the `root` private key, and is what's
+    // used by validators for signing when a keypair is generated from a seed.
+    i++;
+  } while (!curve.r.greaterEquals(privateGen));
 
-  var sec;
-  var key_pair;
-
-  do {
+  var secret;
 
+  if (root) {
+    // As used by validation_create
+    secret = privateGen;
+  } else {
+    publicGen = curve.G.mult(privateGen);
     i = 0;
-
+    // A seed can generate many keypairs as a function of the seed and a uint32.
+    // Almost everyone just uses the first account, `0`,
+    var accountIndex = opts.accountIndex || 0;
     do {
-      sec = sjcl.bn.fromBits(
-              firstHalfOfSHA512(
-                  append_int(
-                    append_int(public_gen.toBytesCompressed(), account_number)
-                    ,
-                    i
-                  )));
+      // We hash the root key-pair's public key bytes, along with the account
```

Comment:
This seems like a really low-level operation. It would probably take me a while to check if this loop is guaranteed to terminate. Are there any libraries that do something like this?